tests/gauss_newton_test.py

```diff
@@ -69,6 +69,21 @@ def setUp(self):
     ])
     self.init_temperature_record_coeffs = onp.array([10, 0.5, 10.5, 50])
 
+  def test_aux_true(self):
```

Comment:
Make it a one step verification (like max_iter =2) with a fixed aux for any state (like True) and check if the aux is still the same. I think other solverrs have such tests. The current tests are already taking a long time so it's preferable to minimize the lengths of the tests if we can.